org.mwc.cmap.TimeController/src/org/mwc/cmap/TimeController/views/AnimatedGif.java

```diff
@@ -0,0 +1,165 @@
+import org.eclipse.swt.graphics.GC;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageLoader;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.mwc.cmap.TimeController.TimeControllerPlugin;
+import org.mwc.cmap.core.CorePlugin;
+
+/**
+ * does the actual animation
+ * @author Ayesha
+ *
+ */
+public class AnimatedGif
+{
+  private static ImageLoader loader = new ImageLoader();
+  private Thread animateThread;
+  private Image image;
+  private ImageData[] imageDataArray;
+  private Label lbl;
+  private boolean cancel;
+  public AnimatedGif(Label lbl,String filename) {
+    this.lbl = lbl;
+    try
+    {
+      imageDataArray = loader.load(FileLocator.openStream(TimeControllerPlugin.getDefault().getBundle(), new Path(filename), false));
+    }
+    catch (IOException e)
+    {
+      CorePlugin.logError(IStatus.ERROR, "Could not find pulsating gif", e);
+    }
+  }
+
+  public void animate() {
+    final Display display = lbl.getDisplay();
+    final GC shellGC = new GC(lbl);
+    final Color shellBackground = lbl.getBackground();
+    final Rectangle bounds = lbl.getBounds();
+    //when animate is called, restart thread
+    cancel=false;
+    if(imageDataArray.length>1) {
+      animateThread = new Thread("Animate") {
+        @SuppressWarnings("unused")
+        public void run() {
+          //run only if not cancel.
+          if(!cancel) {
+            Image offScreenImage = new Image(display, loader.logicalScreenWidth, loader.logicalScreenHeight);
+            GC offScreenImageGC = new GC(offScreenImage);
+            offScreenImageGC.setBackground(shellBackground);
+            offScreenImageGC.fillRectangle(0, 0, loader.logicalScreenWidth, loader.logicalScreenHeight);
+
+            try {
+              /* Create the first image and draw it on the off-screen image. */
+              int imageDataIndex = 0;  
+              ImageData imageData = imageDataArray[imageDataIndex];
+
+              if (image != null && !image.isDisposed()) image.dispose();
+              image = new Image(display, imageData);
+              offScreenImageGC.drawImage(
+                  image,
+                  0,
+                  0,
+                  imageData.width,
+                  imageData.height,
+                  imageData.x,
+                  imageData.y,
+                  imageData.width,
+                  imageData.height);
+
+              /* Now loop through the images, creating and drawing each one
+               * on the off-screen image before drawing it on the shell. */
+              int repeatCount = loader.repeatCount;
+              while (loader.repeatCount == 0 || repeatCount > 0) {
+                switch (imageData.disposalMethod) {
+                  case SWT.DM_FILL_BACKGROUND:
+                    /* Fill with the background color before drawing. */
```

Comment:
@ayeshare - the static checker is flagging an issue in this block. At line 92 we check if bgColor is non-null. But, in the next line we assign it to null, then don't change it.

So, at line 92 it can't be non-null.  I can't remember where this animation code came from, but can you check that we haven't introduced an issue?

Thanks.